# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/windows/KeyboardLayout.cpp
# Commit: 46bf06a42ba6
# Full Hash: 46bf06a42ba623133cfdaceac9d3f9f778975b79
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-03-26 15:34:27
# Description:
#   Bug 1589497 - Make PendingAction of TSFStore copy of key message r=m_kato
#   
#   `TSFTextStore::sHandlingKeyMsg` refers pointer of struct, but referred via
#   `TSFTextStore::PendingAction` so that we should make it has a copy of
#   `sHandlingKeyMsg` because of for async handling.
# ==============================================================================

diff -r 1fe283e3bb59 -r 46bf06a42ba6 widget/windows/KeyboardLayout.cpp
--- a/widget/windows/KeyboardLayout.cpp	Thu Mar 26 07:18:23 2020 +0000
+++ b/widget/windows/KeyboardLayout.cpp	Thu Mar 26 07:33:37 2020 +0000
@@ -1560,8 +1560,10 @@
           ComputeVirtualKeyCodeFromScanCodeEx();
       NS_ASSERTION(mVirtualKeyCode, "Failed to compute virtual keycode");
       break;
-    default:
-      MOZ_CRASH("Unsupported message");
+    default: {
+      MOZ_CRASH_UNSAFE_PRINTF("Unsupported message: 0x%04X", mMsg.message);
+      break;
+    }
   }
 
   if (!mVirtualKeyCode) {